cmd/sonobuoy/app/pluginList.go

```diff
@@ -71,19 +74,53 @@ func (p *pluginList) Set(str string) error {
 	case pluginSystemdLogs:
 		p.DynamicPlugins = append(p.DynamicPlugins, str)
 	default:
-		b, err := ioutil.ReadFile(str)
+		finfo, err := os.Stat(str)
 		if err != nil {
-			return errors.Wrapf(err, "unable to read file '%v'", str)
+			return errors.Wrapf(err, "unable to stat %q", str)
 		}
 
-		newPlugin, err := loadManifest(b)
-		if err != nil {
-			return errors.Wrapf(err, "failed to load plugin file '%v'", str)
+		if finfo.IsDir() {
+			return p.loadPluginsDir(str)
 		}
+		return p.loadSinglePlugin(str)
+	}
+
+	return nil
+}
+
+// loadPluginsDir loads every plugin in the given directory. It does not traverse recursively
+// into the directory. A plugin must have the '.yaml' extension to be considered.
+// It returns the first error encountered and stops processing.
+func (p *pluginList) loadPluginsDir(dirpath string) error {
+	files, err := ioutil.ReadDir(dirpath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read directory %q", dirpath)
+	}
+
+	for _, file := range files {
+		if !file.IsDir() && strings.HasSuffix(file.Name(), fileExtensionYAML) {
+			if err := p.loadSinglePlugin(filepath.Join(dirpath, file.Name())); err != nil {
+				return errors.Wrapf(err, "failed to load plugin in file %q", file.Name())
+			}
+		}
+	}
+
+	return nil
+}
+
+// loadSinglePlugin loads a single plugin located at the given path.
+func (p *pluginList) loadSinglePlugin(filepath string) error {
```

Comment:
This entire section is just moving code around. The existing `load plugin` logic was moved here and then the `load dir of plugins` just gets a list of files and leverages this same method.